cmd/csi-provisioner/csi-provisioner.go

```diff
@@ -180,27 +180,45 @@ func main() {
 		klog.Fatalf("Error getting server version: %v", err)
 	}
 
-	metricsManager := metrics.NewCSIMetricsManager("" /* driverName */)
+	metricsManager := metrics.NewCSIMetricsManagerWithOptions("" /* driverName */)
 
 	grpcClient, err := ctrl.Connect(*csiEndpoint, metricsManager)
 	if err != nil {
 		klog.Error(err.Error())
 		os.Exit(1)
 	}
 
-	err = ctrl.Probe(grpcClient, *operationTimeout)
```

Comment:
We should probably keep it to make sure there is a connection? And add another this check inside the new grpc connection creation block

Reply:
I haven't removed it. Just moved it below so that it used the updated grpc connection. Do you mean that we should have it for both the old and the new connection?

Reply:
Yeah, I think we should have both

Reply:
Ok. I am not sure(couldn't find much in the documentation) how the probe will be handled after this connection is closed. Do you think it will stop making the probe calls to that connection on its own once the connection is closed?

Reply:
Oh, sorry my bad.

https://github.com/kubernetes-csi/external-provisioner/blob/f937d0d4a1d88d866d95fad51148b686d9357cfc/pkg/controller/controller.go#L276

It is probing forever instead of once. While I guess we can just have the new one

Reply:
I took another look at the probeforever implementation. And find that this is a blocking function that will wait for the connection to be established. So I think we should have both the old probe and new probe. Otherwise `provisionerName, err := ctrl.GetDriverName(grpcClient, *operationTimeout)` could fail I guess? 

https://github.com/kubernetes-csi/external-provisioner/blob/f937d0d4a1d88d866d95fad51148b686d9357cfc/vendor/github.com/kubernetes-csi/csi-lib-utils/rpc/common.go#L126

Please also check the code and see if my understanding is correct